### PR TITLE
Merge branch `Davi0kProgramsThings:master` into branch `bitfinexcom:v3-beta`.

### PR DESCRIPTION
**Pull request by Davi0kProgramsThings**
Release notes to attach to the v3.0.0b1 beta release (and tag):
~~~markdown
## bitfinex-api-py v3.0.0b1 release notes (February 23, 2023)
_Author of these release notes: Davi0k (https://github.com/Davi0k)_

bitfinex-api-py v3-beta launches today with v3.0.0b1!

v3-beta simplify the integration of the Bitfinex API in Python 3.8+ applications.

v3.0.0b1 introduces multiple new features, improvements and bug fixes such as:
* full type-hinting and type-checking support with mypy;
* Lots of bug fixes (including critical bugs) and performance boosts;
* a new websocket client to ensure fast, secure and persistent connections;
* a more user friendly implementation of REST endpoints (with support for 75+ REST endpoints);

A complete CHANGELOG of the changes introduced by v3.0.0b1 will be released soon.

> DISCLAIMER:
Production use of v3.0.0b1 (and all future beta versions) is HIGHLY discouraged.
Beta versions should not be used in applications which require user authentication.
Provide your API-KEY/API-SECRET, and manage your account and funds at your own risk.

### How to install bitfinex-api-py v3.0.0b1

bitfinex-api-py v3.0.0b1 can be installed using pip:
```bash
python3 -m pip install bitfinex-api-py==3.0.0b1
```

### A simple ready-to-run example

<ins>[**More v3-beta examples can be found here.**](https://github.com/bitfinexcom/bitfinex-api-py/tree/v3-beta/examples)</ins>

```python
from bfxapi import Client, PUB_WSS_HOST
from bfxapi.websocket import subscriptions
from bfxapi.websocket.types import TradingPairTicker

# Creates a new client connected to https://api-pub.bitfinex.com/v2.
# API-KEY and API-SECRET have not been provided, so only public resources will be available.
# When the client doesn't need to authenticate users, it's always better to use PUB_REST_HOST and PUB_WSS_HOST.
bfx = Client(WSS_HOST=PUB_WSS_HOST)

# Registers an event listener for the t-ticker-update event.
# t-ticker-update event is emitted when updates on trading pairs tickers are received (e.g. tBTCUSD).
# f-ticker-update event is emitted when updates on funding currencies tickers are received (e.g. fUSD).
@bfx.wss.on("t_ticker_update")
def on_t_ticker_update(subscription: subscriptions.Ticker, data: TradingPairTicker):
    print(f"Data: {data}")

# Registers an event listener for the subscribed event.
# The subscribed event is emitted every time a subscription is accepted by the server.
# <subscription> argument contains all the information about the completed subscription.
@bfx.wss.on("subscribed")
def on_subscribed(subscription: subscriptions.Subscription):
    print(f"Subscription with subId <{subscription['subId']}> has been completed.")

# Registers an event listener for the open event.
# The open event will be emitted when the connection is opened.
# Event listeners can be coroutines if they need to handle asyncio operations.
@bfx.wss.once("open")
async def on_open():
    # Subscribes the client to tBTCUSD's ticker public channel.
    await bfx.wss.subscribe("ticker", symbol="tBTCUSD")

# Starts the websocket client with a single connection (default is 5).
# Each connection can handle up to 25 subscriptions to public channels.
# For example, if you know that your application will subscribe to 175 public channels,
#     you will have to use 175 / 25 = 7 connections to handle all subscriptions.
# The default number of connections is 5; so, if the <connections> argument is not provided,
#     the client will be able to handle a maximum of 25 * 5 = 125 subscriptions.
# Keep in mind that using a large number of connections could slow down the client performance,
#     the use of more than 20 connections is strongly discouraged.
bfx.wss.run(connections=1)
```
~~~

### CODE REVIEW

File: bfxapi/enums.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ERR_AUTH_HMAC = 10113
+    ERR_AUTH_NONCE = 10114
+    ERR_UNAUTH_FAIL = 10200
+    ERR_SUB_FAIL = 10300
+    ERR_SUB_MULTI = 10301
+    ERR_SUB_UNK = 10302
+    ERR_SUB_LIMIT = 10305
+    ERR_UNSUB_FAIL = 10400
+    ERR_UNSUB_NOT = 10401
+    ERR_READY = 11000
```
Comment (vigan-abd) [inline bfxapi/enums.py:50]: one thing, please configure your IDE to append newline to end of each file as it's linux standard

File: bfxapi/urls.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+REST_HOST = "https://api.bitfinex.com/v2"
+PUB_REST_HOST = "https://api-pub.bitfinex.com/v2"
+STAGING_REST_HOST = "https://api.staging.bitfinex.com/v2"
```
Comment (vigan-abd) [inline bfxapi/urls.py:3]: please remove all staging related urls

File: bfxapi/utils/logger.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    def format(self, record):
+        levelname = record.levelname
+        if self.use_color and levelname in COLORS:
+            levelname_color = COLOR_SEQ % (30 + COLORS[levelname]) + levelname + RESET_SEQ
+            record.levelname = levelname_color
+        record.name = ITALIC_COLOR_SEQ % (30 + BLACK) + record.name + RESET_SEQ
+        return logging.Formatter.format(self, record)
+
+class ColoredLogger(logging.Logger):
```
Comment (vigan-abd) [inline bfxapi/utils/logger.py:40]: do we support also non colored logging, e.g. it's useful when you want to pipe stdout to a file

File: examples/rest/merchant/submit_invoice.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# python -c "import examples.rest.merchant.submit_invoice"
```
Comment (vigan-abd) [inline examples/rest/merchant/submit_invoice.py:1]: can we add later support for the rest of merchant endpoints please? it's really useful for 3rd parties
https://docs.bitfinex.com/reference/merchant-endpoints

File: pylint.rc [hunk 5]
```diff
@@ -1,13 +0,0 @@
-[MESSAGES CONTROL]
```
Comment (vigan-abd) [inline pylint.rc:1]: let's add support for pylint, we need to have same syntax across all project

**Review by vigan-abd — CHANGES_REQUESTED** [review]

**Review by itsdeka — APPROVED** [review]
LGTM

**Review by vigan-abd — APPROVED** [review]

**Pull request merged**